dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/netty4/h2/NettyHttp2ProtocolSelectorHandler.java

```diff
@@ -68,8 +69,9 @@
             h2StreamChannel = new Http2WriteQueueChannel(h2StreamChannel, writeQueue);
         }
         ChannelPipeline pipeline = ctx.pipeline();
-        pipeline.addLast(
-                new NettyHttp2FrameHandler(h2StreamChannel, factory.newInstance(h2StreamChannel, url, frameworkModel)));
+        Http2TransportListener http2TransportListener = factory.newInstance(h2StreamChannel, url, frameworkModel);
+        ctx.channel().closeFuture().addListener(future -> http2TransportListener.onStreamClosed());
```

Comment:
**Performance Issue**: Using a listener to close streams might introduce a delay in freeing up resources, potentially leading to memory leaks if the streams are not closed in a timely manner. This is critical in a high-throughput system where resources are constantly being allocated and deallocated. <br> **Fix**: Consider implementing a more proactive resource management strategy, such as explicitly closing streams in a finally block or using try-with-resources (for Java 7 and above) to ensure that resources are freed as soon as they are no longer needed. <br> **Code Suggestion**: 
 ```
 try (Stream stream = createStream()) {
     // use the stream
 } catch (Exception e) {
     // handle exception
 }
 ```